bukuserver/response.py

```diff
@@ -1,4 +1,45 @@
-response_template = {
-    "success": {'status': 0, 'message': 'success'},
-    "failure": {'status': 1, 'message': 'failure'}
-}
+from enum import Enum
+from flask import jsonify
+from flask_api import status
+
+OK, FAIL = 0, 1
+
+class Response(Enum):
+    SUCCESS = (status.HTTP_200_OK, "Success.")
+    FAILURE = (status.HTTP_400_BAD_REQUEST, "Failure.")
+    INPUT_NOT_VALID = (status.HTTP_400_BAD_REQUEST, "Request not valid")
+    BOOKMARK_NOT_FOUND = (status.HTTP_404_NOT_FOUND, "Bookmark not found.")
+    TAGS_MISSING = (status.HTTP_400_BAD_REQUEST, "Replacement tags are missing.")
+    TAG_NOT_FOUND = (status.HTTP_404_NOT_FOUND, "Tag not found.")
+    TAGS_NOT_LIST = (status.HTTP_400_BAD_REQUEST, "List of tags expected.")
+    TAG_NOT_VALID = (status.HTTP_400_BAD_REQUEST, "Invalid tag.")
+
+    @staticmethod
+    def from_flag(flag: bool):
+        return Response.SUCCESS() if flag else Response.FAILURE()
+
+    @property
+    def status_code(self):
+        return self.value[0]
+
+    @property
+    def message(self):
+        return self.value[1]
+
+    @property
+    def status(self):
+        return OK if self.status_code == status.HTTP_200_OK else FAIL
+
+    def get_json(self, data :dict = None):
+        json = {'status': self.status, 'message': self.message}
+        if isinstance(data, dict):
+            json.update(data)
+        return json
```

Comment:
Technically this entire function is equivalent to the expression
```py
dict(status=self.status, message=self.message, **data or {})
```
(At least not accounting for the possibility of the data being an invalid value… but one could argue that a `TypeError` would be a more adequate result of an invalid invocation.)

…Also, naming it simply `.json()` might be better (especially since it's a pure function to begin with).

Reply:
My initial intention with this function was to allow overriding the message or status  for specific cases without having to add new items to the Response Enum, e.g.:
```
Response.SUCCESS(data={'message':  "Successfully updated record"})
Response.FAILURE(data={'status':  status.HTTP_403_FORBIDDEN})
```
Although i concede it defeats the purpose of gathering all message strings in the same location. If it's not desired I will update using your solution.

Reply:
…Ah. I haven't considered the option of overriding the core fields.

Though TBH, making them arbitrary doesn't seem like a good idea to me; this file is a part of the web-app code, so if a new response is needed, it can simply be added here (and providing specifics is what metadata is _for_).

Reply:
…Incidentally, using a dict literal instead of `dict()` doesn't result in an error when passing existing keys (dict literals don't seem to have a check for repeating keys).

Reply:
ok, I changed it to dict literal because pylint was complaining